### PR TITLE
Add Docker Sensor

### PR DESCRIPTION
**Pull request by bravefoot**
Add Docker Sensor that succeeds based on whether a Docker container successfully exits.

Also moves a lot of logic for running docker containers from DockerOperator to a new hook class.

### CODE REVIEW

**Comment by potiuk** [comment]
Some statoc checks and docs are failing

File: airflow/providers/docker/hooks/docker.py [hunk 1]
```diff
@@ -17,13 +17,70 @@
...
 
-from docker import APIClient
+from docker import APIClient, tls
 from docker.errors import APIError
 
 from airflow.exceptions import AirflowException
 from airflow.hooks.base import BaseHook
 from airflow.utils.log.logging_mixin import LoggingMixin
 
+"""
```
Comment (mik-laj) [inline airflow/providers/docker/hooks/docker.py]: Docstring should be in method/class, not above it.

Comment (bravefoot) [reply airflow/providers/docker/hooks/docker.py]: Thanks for the tip. I moved it.

**Comment by potiuk** [comment]
Seems like tests need fixing @bravefoot

**Pull request closed without merging**